packages/frontend/src/retro/hooks/useTimer.ts

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useMemo, useState } from "react";
+
+interface useTimerProps {
+  timerFinished: () => void;
+  defaultDuration: number;
+}
+export function useTimer({ timerFinished, defaultDuration }: useTimerProps) {
+  const [isRunning, setIsRunning] = useState(false);
+  const [duration, setDuration] = useState(defaultDuration);
+  const finishTime = useMemo(() => Date.now() + duration, [duration]);
+  const [milliseconds, setMilliseconds] = useState(finishTime - Date.now());
+  const minutes = getMinutes(milliseconds);
+  const seconds = getSeconds(milliseconds);
+
+  function startTimer(duration: number) {
+    console.log("start timer with {} duration", duration);
+    console.log("current time is {}", Date.now());
+    setDuration(duration);
+    setMilliseconds(duration);
+    setIsRunning(true);
+  }
+  function stopTimer() {
+    console.log("stop timer");
+    setIsRunning(false);
+  }
+  function pauseTimer() {
+    console.log("pause timer");
+    setIsRunning(false);
+  }
+
+  function createLabel(ms: number) {
+    return new Date(ms).toISOString().slice(14, 19);
+  }
+  function getMinutes(ms: number) {
+    console.log("minutes {}", Math.floor(ms / 1000 / 60));
+    return Math.floor(ms / 1000 / 60);
+  }
+  function getSeconds(ms: number) {
+    console.log("seconds {}", (ms / 1000) % 60);
+    return (ms / 1000) % 60;
+  }
+  useEffect(() => {
+    function endTimer() {
+      console.log("end timer");
+      setIsRunning(false);
+      setDuration(-1);
```

Comment:
I'd rather use undefined here

Reply:
The duration is currently defined as a strict number, as i understand it typescript does not allow undefined as value. But i see that a value of -1 or undefined is not the way to go. I think there should be an default value to set the timer to the last chosen timer length

Reply:
This will be resolved once it's set to the last timer.